src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -1,22 +1,186 @@
+  const endpoints = rewardsAttestationEndpoints?.split(',') || null
+  const hasConfig =
+    oracleEthAddress && AAOEndpoint && quorumSize && quorumSize > 0
 
-const NATIVE_MOBILE = process.env.REACT_APP_NATIVE_MOBILE
+  if (!hasConfig) {
+    console.error('Error claiming rewards: Config is missing')
+    return
+  }
+  try {
+    const response: { error?: string } = yield call(
+      AudiusBackend.submitAndEvaluateAttestations,
+      {
+        challengeId,
+        encodedUserId: encodeHashId(currentUser.user_id),
+        handle: currentUser.handle,
+        recipientEthAddress: currentUser.wallet,
+        specifier,
+        oracleEthAddress,
+        amount,
+        quorumSize,
+        endpoints,
+        AAOEndpoint
+      }
+    )
+    if (response.error) {
+      if (retryOnFailure) {
+        yield put(claimChallengeRewardWaitForRetry(claim))
+        switch (response.error) {
+          case FailureReason.HCAPTCHA:
+            // Hide the Challenge Rewards Modal because the HCaptcha modal doesn't look good on top of it.
+            // Will be restored on close of the HCaptcha modal.
+            yield put(
+              setVisibility({
+                modal: CHALLENGE_REWARDS_MODAL_NAME,
+                visible: false
+              })
+            )
+            yield put(
+              setVisibility({ modal: HCAPTCHA_MODAL_NAME, visible: true })
+            )
+            break
+          case FailureReason.COGNITO_FLOW:
+            // TODO: Uncomment once Cognito Modal is added
+            // yield put(
+            //   setVisibility({ modal: COGNITO_MODAL_NAME, visible: true })
+            // )
+            break
+          case FailureReason.BLOCKED:
+            throw new Error('User is blocked from claiming')
+          case FailureReason.UNKNOWN_ERROR:
+          default:
+            throw new Error(`Unknown Error: ${response.error}`)
+        }
+      } else {
+        yield put(claimChallengeRewardFailed())
+      }
+    } else {
+      yield put(
+        increaseBalance({
+          amount: stringAudioToStringWei(amount.toString() as StringAudio)
+        })
+      )
+      yield put(setUserChallengeDisbursed({ challengeId }))
+      yield put(claimChallengeRewardSucceeded())
+    }
+  } catch (e) {
+    console.error('Error claiming rewards:', e)
+    yield put(claimChallengeRewardFailed())
+  }
+}
+
+function* watchSetHCaptchaStatus() {
+  yield takeLatest(setHCaptchaStatus.type, function* (
+    action: ReturnType<typeof setHCaptchaStatus>
+  ) {
+    const { status } = action.payload
+    yield call(retryClaimChallengeReward, status === HCaptchaStatus.SUCCESS)
+  })
+}
+
+function* watchSetCognitoFlowStatus() {
+  yield takeLatest(setCognitoFlowStatus.type, function* (
+    action: ReturnType<typeof setCognitoFlowStatus>
+  ) {
+    const { status } = action.payload
+    // Only attempt retry on closed, so that we don't error on open
+    if (status === CognitoFlowStatus.CLOSED) {
```

Comment:
what if it's closed on a cognito flow failure, would this still retry the claim?

Reply:
I think it should - In the modal, `flow.close()` is called in the error handler